Add suggestions

sdodson opened the pull request:
This adds minor_min, minor_block_list, minor_max, z_min, z_block_list,
and z_max which define the minimum, a list of blocked versions, and a
maximum version which can be used to filter upgrade edges. It's expected
that this is applied atop a list of versions already present in a
channel so the block_list may not see much use.

There's a default as well as arch specific definitions. If you define
arch specific data you are expected to define the entire structure.

sdodson commented:
To be used with https://github.com/openshift/cincinnati-graph-data/pull/425
Not sure if we should make these arch specific or not?

sdodson commented:
/wip

jupierce commented:
Having the files include a default with an optional arch override might be easier to maintain.
```
default:
  minor_floor: 4.2.21
  minor_block_list: []
  z_floor: 4.3.0
  z_block_list: [] 
s390x:
  minor_floor: 4.2.26
```

sdodson commented:
/wip cancel
@jupierce PTAL

LalatenduMohanty commented:
/lgtm 

/hold till we get a review from @jupierce

sdodson commented:
/hold cancel
I'd like to move forward with this provided everyone agrees the fields I've defined here provide everything we'd need. I added the max versions so that we can accommodate 4.6 EUS cycle which may potentially have scenarios where we stop shipping 4.7 for a few months even if we're tagging new builds. Since that exact mechanic is as of yet undefined I wanted to provide this flexibility. I've updated https://github.com/openshift/doozer/pull/277 to also support maxes.

LalatenduMohanty commented:
/lgtm
